### PR TITLE
Re-implement stacktrace

### PR DESCRIPTION
**Pull request by snnn**
### Description
Re-implement stacktrace. The new implementation doesn't directly use Windows API, hence can avoid problems regarding to initialize/uninitialize the dbghelp library.

### CODE REVIEW

**Comment by pranavsharma** [comment]
Does this mean we'll need C++23 to compile ORT on Windows? If the default logger is the real issue, we can do 1 of 2 things: 1. have a truly default logger that relies on function scoped static instance 2. replace the usage of logger in this code with cerr?

**Comment by snnn** [comment]
On Windows we only use VC++ as the compiler, and it has great support for C++23. We shouldn't worry about it.

File: onnxruntime/core/platform/windows/stacktrace.cc [hunk 1]
```diff
@@ -46,103 +40,19 @@ std::vector<std::string> GetStackTrace() {
...
-  // hide CaptureStackTrace::Trace and GetStackTrace so the output starts with the 'real' location
-  constexpr int frames_to_skip = 2;
-
-  // we generally want to skip the first two frames, but if something weird is going on (e.g. code coverage is
-  // running) and we only have 1 or 2 frames, output them so there's at least something that may be meaningful
-  const uint16_t start_frame = num_frames > frames_to_skip ? frames_to_skip : 0;
-  for (uint16_t i = start_frame; i < num_frames; ++i) {
-    stacktrace.push_back(Lookup(f[i]));
+  auto st = std::stacktrace::current();
+  for (auto& stack : st) {
```
Comment (pranavsharma) [inline onnxruntime/core/platform/windows/stacktrace.cc:47]: const auto&

**Review by pranavsharma — APPROVED** [review]

**Review by pranavsharma — APPROVED** [review]

**Comment by skottmckay** [comment]
Do we still need to support building with VS 2019?

**Comment by snnn** [comment]
> Do we still need to support building with VS 2019?

I am not aware of such a need. Compare to GCC,  Visual Studio is easy to upgrade. It doesn't bound to OS version.  You can install the latest VS on a very old Windows version and build applications that still target old Windows versions.

**Pull request merged**